src/api/modelTypes.js

```diff
@@ -117,6 +117,10 @@ export type CrossRealmBot = {|
   user_id: number,
 
   // timezone included since commit 58ee3fa8c (in 1.9.0)
+  // Currently, bots have a `timezone` field of ''. In future, as Tim mentions
+  // at https://github.com/zulip/zulip-mobile/pull/3789#issuecomment-581218576,
+  // that field may be omitted to reduce payload sizes. So timezone is correctly
+  // an optional property in CrossRealmBot, but correctly required in User.
   timezone?: string,
```

Comment:
"Currently" is perilous in comments. :stuck_out_tongue: The future may eventually come along, and when it does the comment will tend to still be there and so will be wrong.

A simple fix is to say "As of 2020-02".

It'd also be good to integrate what this comment is saying and what the existing comment says, as they're part of the same story (the behavior of this field through the server's history.) I think it amounts to: the `timezone` field is included since that commit, but at least as of 2020-02, it's always `''` for bots. And so as Tim mentions at that link, in the future it may be omitted etc.

Reply:
Oh, one thing that's a bit subtle (as our terminology is confusing) that I see reading @timabbott's comment again: he's talking about *bots*, and that's a wider category than *cross-realm bots*. The latter are a fixed handful like Welcome Bot. The former can be created by users in a given realm... and they show up as values of type `User`. So I think in fact when it comes to this possible future, there isn't a difference between `CrossRealmBot` and `User`, because the latter includes bots too.

Which I guess means it'd be good to make it optional in `User`! That needn't block these other changes, though -- that's just some future-proofing based on Tim's comment in this thread. Should in any case make the comment reflect that, though.

Reply:
Also just added some jsdoc which hopefully clarifies this some: 60807bde2